### PR TITLE
Move wal queue to its own repository and minor bug fixes.

### PR DESCRIPTION
**Pull request by mattdurham**
This moves the queue implementation to a separate repo github.com/grafana/walqueue. In that repo I made changes to support reloading configuration and preventing duplicate metric registration on config relods.

### CODE REVIEW

**Comment by mattdurham** [comment]
Setting this back to draft, since I am going to move this functionality into a fork.

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ Main (unreleased)
 
 - Fixed an issue in the `prometheus.exporter.postgres` component that would leak goroutines when the target was not reachable (@dehaansa)
 
+- Moved the wal queue implementation to a separate repository to share with other projects. Also fixed issue with reloading
+  configuration and prometheus duplication of metrics. (@mattdurham)
```
Comment (thampiotr) [inline CHANGELOG.md:26]:
```suggestion
- Fixed issue with reloading configuration and prometheus metrics duplication in `prometheus.write.queue`. (@mattdurham)
```

Comment (thampiotr) [reply CHANGELOG.md:26]: I don't think it's user-facing that we moved to use a library.

**Review by thampiotr — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -940,3 +941,6 @@ exclude (
 )
 
 replace github.com/prometheus/procfs => github.com/prometheus/procfs v0.12.0
+
+// This is to handle issues witn synchronous mailbox and closing channels.
+replace github.com/vladopajic/go-actor => github.com/grafana/go-actor v0.0.0-20241113133736-e18c4a5c12f4
```
[the remark this answers is not in the document]
Comment (mattdurham) [reply go.mod:946]: Working with them  upstream as we speak.

**Comment by mattdurham** [comment]
This also fixes #2074

**Pull request merged**